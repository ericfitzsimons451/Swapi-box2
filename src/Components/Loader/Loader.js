import React from 'react'
import './Loader.scss'

const Loader = () => {
    return (
        <div className="loader">
            <h1>SwapiBox</h1>
            <h2>Content loading...</h2>
            <h3>May the <span>Force</span> be with you!</h3>
        </div>
    )
}

export default Loader