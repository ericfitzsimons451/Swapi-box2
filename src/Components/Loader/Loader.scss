.loader {
    background-image: url('https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRk1JPVMTAdJymKTf3W62Xdd2U6VWRk9FNxLsvRATrWBER1BHmN');
    min-height: 100vh;
    background-size: cover;
    overflow: hidden;
}

.loader h1,
.loader h2,
.loader h3 {
    font-size-adjust: 100;
    position: relative;
    left: 10vw;
    top: 10vh;
    color: black;
}
.loader span {
    color: purple;
}