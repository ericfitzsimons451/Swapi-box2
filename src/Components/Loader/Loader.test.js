import React from 'react'
import { shallow } from 'enzyme'
import Loader from '../Loader/Loader'

describe('Loader', () => {
    it('renders the component with loading text', () => {
        const wrapper = shallow(<Loader />)
        const title = <h1>SwapiBox</h1>
        const loadText1 = <h2>Content loading...</h2>
        const loadText2 = <h3>May the <span>Force</span> be with you!</h3>

        expect(wrapper.contains(title, loadText1, loadText2)).toEqual(true)
    })

    it('should match the snapshot with no data passed down', () => {
        const wrapper = shallow(<Loader />)
        expect(wrapper).toMatchSnapshot()
    })
})