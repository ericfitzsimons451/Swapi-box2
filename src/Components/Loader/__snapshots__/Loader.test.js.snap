// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match the snapshot with no data passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SwapiBox
        </h1>,
        <h2>
          Content loading...
        </h2>,
        <h3>
          May the 
          <span>
            Force
          </span>
           be with you!
        </h3>,
      ],
      "className": "loader",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SwapiBox",
        },
        "ref": null,
        "rendered": "SwapiBox",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Content loading...",
        },
        "ref": null,
        "rendered": "Content loading...",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "May the ",
            <span>
              Force
            </span>,
            " be with you!",
          ],
        },
        "ref": null,
        "rendered": Array [
          "May the ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Force",
            },
            "ref": null,
            "rendered": "Force",
            "type": "span",
          },
          " be with you!",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SwapiBox
          </h1>,
          <h2>
            Content loading...
          </h2>,
          <h3>
            May the 
            <span>
              Force
            </span>
             be with you!
          </h3>,
        ],
        "className": "loader",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SwapiBox",
          },
          "ref": null,
          "rendered": "SwapiBox",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Content loading...",
          },
          "ref": null,
          "rendered": "Content loading...",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "May the ",
              <span>
                Force
              </span>,
              " be with you!",
            ],
          },
          "ref": null,
          "rendered": Array [
            "May the ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Force",
              },
              "ref": null,
              "rendered": "Force",
              "type": "span",
            },
            " be with you!",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
